src/HPWH.cc

```diff
@@ -5322,3 +5378,565 @@ int HPWH::HPWHinit_file(string configFile)
+                {
+                    totalDrawVolume_L += drawVolume_L;
+                    drawVolume_L = 0.;
+                    isDrawing = false;
+                    drMode = DR_ALLOW;
+                    maxTankT_C = tankT_C;        // initialize for next pass
+                    maxOutletT_C = outletTemp_C; // initialize for next pass
+                    prevAvgOutletT_C = avgOutletT_C;
+                    prevMinOutletT_C = minOutletT_C;
+                    ++step;
+                }
+            }
+            break;
+        }
+
+        case 1:
+        {
+            if (isHeating) // ensure heat is on before proceeding
+            {
+                ++step;
+            }
+            break;
+        }
+
+        case 2: // heating
+        {
+            if ((tankT_C > maxTankT_C) && isHeating &&
+                (elapsedTime_min <
+                 60)) // has not reached maxTankT, heat is on, and less than 1 hr has elpased
+            {
+                maxTankT_C = std::max(tankT_C, maxTankT_C);
+            }
+            else // start another draw
+            {
+                firstDraw = false;
+                isDrawing = true;
+                drawVolume_L = 0.;
+                drMode = DR_LOC;
+                step = 0; // repeat
+            }
+        }
+        }
+
+        drawVolume_L += incrementalDrawVolume_L;
+        ++elapsedTime_min;
+    }
+
+    //
+    if (totalDrawVolume_L < GAL_TO_L(18.))
+    {
+        usage = Usage::VerySmall;
+    }
+    else if (totalDrawVolume_L < GAL_TO_L(51.))
+    {
+        usage = Usage::Low;
+    }
+    else if (totalDrawVolume_L < GAL_TO_L(75.))
+    {
+        usage = Usage::Medium;
+    }
+    else
+    {
+        usage = Usage::High;
+    }
+
+    return true;
+}
+
+//-----------------------------------------------------------------------------
+///	@brief	Performs standard 24-hr test
+/// @note	see https://www.regulations.gov/document/EERE-2019-BT-TP-0032-0058
+/// @param[in/out] standardTestSummary	 specifies usage on input, test metrics on output
+///	@param[in]	setpointT_C		         setpoint temperature (optional)
+/// @return	true (success), false (failure).
+//-----------------------------------------------------------------------------
+bool HPWH::run24hrTest(const Usage& usage,
+                       StandardTestSummary& standardTestSummary,
+                       const double setpointT_C /* = 51.7 */)
+{
+    // select the draw pattern based on usage
+    DrawPattern* drawPattern = nullptr;
+    switch (usage)
+    {
+    case Usage::VerySmall:
+    {
+        drawPattern = &verySmallUsage;
+        break;
+    }
+    case Usage::Low:
+    {
+        drawPattern = &lowUsage;
+        break;
+    }
+    case Usage::Medium:
+    {
+        drawPattern = &mediumUsage;
+        break;
+    }
+    case Usage::High:
+    {
+        drawPattern = &highUsage;
+        break;
+    }
+    }
+    if (drawPattern == nullptr)
+    {
+        return false;
+    }
```

Comment:
This could all be replaced with a single line if we use an `unordered_map` per suggestion above.